### PR TITLE
add comments to QuestFacotry

### PR DESCRIPTION
**Pull request by waynehoover**

### CODE REVIEW

File: contracts/QuestFactory.sol [hunk 1]
```diff
@@ -54,6 +56,17 @@ contract QuestFactory is Initializable, OwnableUpgradeable, AccessControlUpgrade
         protocolFeeRecipient = protocolFeeRecipient_;
     }
 
+    /// @dev Create either an erc20 or erc1155 quest
```
Comment (GarrettJMU) [inline contracts/QuestFactory.sol:59]: we should also add that it's whitelisted deployment only

**Review by GarrettJMU — APPROVED** [review]

File: contracts/QuestFactory.sol [hunk 2]
```diff
@@ -115,45 +128,61 @@ contract QuestFactory is Initializable, OwnableUpgradeable, AccessControlUpgrade
...
         _revokeRole(CREATE_QUEST_ROLE, account_);
     }
 
+    /// @dev grant the default admin role and the create quest role to the owner
     function grantDefaultAdminAndCreateQuestRole() public onlyOwner {
         _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
         _grantRole(CREATE_QUEST_ROLE, msg.sender);
     }
 
+    /// @dev set the claim signer address
```
Comment (jonathandiep) [inline contracts/QuestFactory.sol:149]: Should we also add `@param` for address? (same with functions below)

File: contracts/QuestFactory.sol [hunk 3]
```diff
@@ -115,45 +128,61 @@ contract QuestFactory is Initializable, OwnableUpgradeable, AccessControlUpgrade
...
     function recoverSigner(bytes32 hash, bytes memory signature) public pure returns (address) {
         bytes32 messageDigest = keccak256(abi.encodePacked('\x19Ethereum Signed Message:\n32', hash));
         return ECDSAUpgradeable.recover(messageDigest, signature);
     }
 
-    // This contract must be set as Minter on the receipt contract.
+    /// @dev mint a RabbitHole Receipt. Note: this contract must be set as Minter on the receipt contract
+    /// @param questId_ The id of the quest
+    /// @param hash_ The hash of the message
+    /// @param signature_ The signature of the message
```
Comment (jonathandiep) [inline contracts/QuestFactory.sol:185]: Should be "The signature of the hash"

File: contracts/QuestFactory.sol [hunk 4]
```diff
@@ -115,45 +128,61 @@ contract QuestFactory is Initializable, OwnableUpgradeable, AccessControlUpgrade
...
     function setRewardAllowlistAddress(address rewardAddress_, bool allowed_) public onlyOwner {
         rewardAllowlist[rewardAddress_] = allowed_;
     }
 
+    /// @dev return the number of minted receipts for a quest
     function getNumberMinted(string memory questId_) external view returns (uint) {
         return quests[questId_].numberMinted;
     }
 
+    /// @dev recover the signer from a hash and signature
```
Comment (jonathandiep) [inline contracts/QuestFactory.sol:176]: Params for hash & signature?

File: contracts/QuestFactory.sol (hunk 2 above)
Comment (waynehoover) [reply contracts/QuestFactory.sol:149]: yes, thanks added.

File: contracts/QuestFactory.sol (hunk 3 above)
Comment (waynehoover) [reply contracts/QuestFactory.sol]: yes.

File: contracts/QuestFactory.sol (hunk 1 above)
Comment (waynehoover) [reply contracts/QuestFactory.sol]: done.

**Review by jonathandiep — APPROVED** [review]

File: contracts/QuestFactory.sol (hunk 4 above)
Comment (waynehoover) [reply contracts/QuestFactory.sol:180]: yep. thanks.

**Pull request merged**